src/validation/statement.rs

```diff
@@ -287,13 +287,34 @@ fn validate_for_loop<T: AnnotationMap>(
     //       by a VAR_INPUT {ref} function call.
 }
 
+fn validate_if_statement<T: AnnotationMap>(
+    validator: &mut Validator,
+    context: &ValidationContext<T>,
+    statement: &IfStatement,
+) {
+    for block in &statement.blocks {
+        let kind = context.annotations.get_type_or_void(&block.condition, context.index);
+
+        if !kind.get_type_information().is_bool() {
```

Comment:
Not that this is wrong, but i'm interested to see how often people also use integers here the C way.. 
I would just accept this for now and and we discuss later if it might need to become a warning. In which case it will need its own error code btw.
Also note to self, when merging the diagnostics make sure this becomes E094